File: LeetCode/Easy/0001_Two_Sum/Solution.go
package main

func twoSum(nums []int, target int) []int {
	answer := []int{0, 0}
	track := map[int]int{}

	for i := 0; i < len(nums); i++ {
		if val, ok := track[nums[i]]; ok {
			answer[0] = i
			answer[1] = val
		} else {
			track[target-nums[i]] = i
		}
	}

	return answer
}

File: LeetCode/Easy/0001_Two_Sum/Solution.cpp
// 2022-05-26
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        for (int i = 0; i < nums.size(); i++) {
            if (seen.count(nums[i])) {
                return {seen[nums[i]], i};
            }
            seen[target - nums[i]] = i;
        }
        return {-1, -1};
    }
};